src/FffGcodeWriter.cpp

```diff
@@ -1869,64 +1869,135 @@ void FffGcodeWriter::processTopBottom(const SliceDataStorage& storage, LayerPlan
             {
                 if (bridge2 > -1)
                 {
-                    // orientate second bridge skin at 90 deg to first
-                    skin_angle = (bridge2 + 90) % 360;
+                    if (bottom_layers > 2)
+                    {
+                        // orientate second bridge skin at +45 deg to first
+                        skin_angle = (bridge2 + 45) % 360;
+                    }
+                    else
+                    {
+                        // orientate second bridge skin at 90 deg to first
+                        skin_angle = (bridge2 + 90) % 360;
+                    }
                 }
-                use_bridge_config2 = true;
                 pattern = EFillMethod::LINES; // force lines pattern on upper bridge skins
-                skin_overlap = std::max(skin_overlap, (coord_t)(mesh_config.insetX_config.getLineWidth() / 2)); // force a minimum amount of skin_overlap
+                skin_overlap = more_skin_overlap;
                 skin_density = mesh.getSettingInPercentage("bridge_skin_density_2") / 100;
+                skin_config = &mesh_config.bridge_skin_config2;
             }
-            else if (layer_nr > 2)
+            else if (layer_nr > 2 && bottom_layers > 2)
             {
-                // if this is the third bridge layer, use the same skin_angle as the first
-                Polygons supportedSkinPartRegions3;
-                int bridge3 = bridgeAngle(skin_part.outline, &mesh.layers[layer_nr - 3], supportedSkinPartRegions3);
-                if (bridge3 > -1)
+                // if this is the third bridge layer use bridge_skin_config3
+                Polygons supported_skin_part_regions3;
+                if (support_layer_nr >= 2)
+                {
+                    support_layer = &storage.support.supportLayers[support_layer_nr - 2];
+                }
+                // outline used is union of current skin part and those skin parts from the 1st bridge layer that overlap the curent skin part
+                Polygons skin3_outline(skin_part.outline);
+                for (auto layer_part : mesh.layers[layer_nr - 2].parts)
                 {
-                    skin_angle = bridge3;
+                    for (auto other_skin_part : layer_part.skin_parts)
+                    {
+                        if (PolygonUtils::polygonsIntersect(skin_part.outline.outerPolygon(), other_skin_part.outline.outerPolygon()))
+                        {
+                            skin3_outline = skin3_outline.unionPolygons(other_skin_part.outline);
+                        }
+                    }
+                }
+                int bridge3 = bridgeAngle(skin3_outline, &mesh.layers[layer_nr - 3], support_layer, supported_skin_part_regions3, support_threshold);
+                if (bridge3 > -1 || (supported_skin_part_regions3.area() / (skin3_outline.area() + 1) < support_threshold))
```

Comment:
Sorry, I don't think that's going to make things much simpler due to the fact that each of the skin layers are handled differently in terms of line config, support areas, etc.

Reply:
I think there's two things that should be done in order to make this code simpler.

The first, and most major change is that we should remove the nesting here. This coincides with the reaction that I'll post shortly: The detection for which bridge layer a polygon is part of is not functional right now, because it puts the entire part on the same "bridge layer". The nesting can be removed because there is nothing inside the `if` statement that depends on the results before them.

I think that once the nesting is removed and the bridge detection is fixed, you'll see that it becomes much easier to factor the 1, 2 and 3 parts out. Of course, the different parameters still need to be different for each call, but then it's also logical that they are parameters of the function call because these are the parameters for bridging in general.